botocore/credentials.py

```diff
@@ -1919,10 +1920,21 @@ def _retrieve_or_fail(self):
         )
 
     def _build_headers(self):
-        auth_token = self._environ.get(self.ENV_VAR_AUTH_TOKEN)
-        if auth_token is not None:
+        auth_token = None
+        if self.ENV_VAR_AUTH_TOKEN_FILE in self._environ:
+            auth_token_file_path = self._environ[self.ENV_VAR_AUTH_TOKEN_FILE]
+            with open(auth_token_file_path) as token_file:
+                auth_token = token_file.read()
+        elif self.ENV_VAR_AUTH_TOKEN in self._environ:
+            auth_token = self._environ[self.ENV_VAR_AUTH_TOKEN]
+        if auth_token:
+            self._validate_auth_token(auth_token)
             return {'Authorization': auth_token}
 
+    def _validate_auth_token(self, auth_token):
+        if "\r\n" in auth_token:
+            raise ValueError("Auth token value is not a legal header value")
```

Comment:
It can be either character, we don't want any newlines/carriage returns in the header. Either of these options is probably sufficiently correct:

```python
_TOKEN_VALIDATOR = re.compile(r"^[^\r\n]*$")
[...]
    def _validate_auth_token(self, auth_token):
        if not _TOKEN_VALIDATOR.match(auth_token):
            raise ValueError("Auth token value is not a legal header value")
```

or just:

```python
if "\r" in auth_token or "\n" in auth_token: